scripts/update_config.py

```diff
@@ -0,0 +1,432 @@
+        model_name = extract_model(path)
+
+        if eval_name and model_name:
+            key = f"{eval_name}:{model_name}"
+            eval_model_paths[key].append(path)
+
+    # Sort by date and get the most recent path for each eval-model combination
+    result = []
+    for _, paths_list in eval_model_paths.items():
+        # Sort by timestamp in the path
+        sorted_paths = sorted(
+            paths_list,
+            key=lambda x: extract_timestamp(x),
+            reverse=True,  # Most recent first
+        )
+        # Take the most recent path
+        if sorted_paths:
+            result.append(sorted_paths[0])
+
+    return result
+
+
+def extract_eval_name(path):
+    """Extract evaluation name from an S3 path."""
+    match = re.search(r"logs/stage/([^/]+)/", path)
+    if match:
+        return match.group(1).lower().replace("-", "_")
+    return None
+
+
+def extract_model(path):
+    """Extract model name from an S3 path."""
+    match = re.search(r"/([^/]+\+[^/]+)/", path)
+    if match:
+        return match.group(1)
+    return None
+
+
+def extract_timestamp(path):
+    """Extract timestamp from path for sorting by recency."""
+    # We support both T and - as a separator
+    match = re.search(r"(\d{4}-\d{2}-\d{2}[T-]\d{2}-\d{2}-\d{2})", path)
+    if match:
+        return match.group(1)
+
+    # Default to empty string if no timestamp found
+    return ""
+
+
+def extract_environment(path):
+    if "logs/stage/" in path:
+        # Stage logs count as both stage and dev
+        return ["stage", "dev"]
+    elif "logs/prod/" in path:
+        return ["prod"]
+    elif "logs/dev/" in path:
+        return ["dev"]
+    return ["dev"]  # Default to dev if not found
+
+
+def get_default_metrics_from_config(original_config, eval_name, env="staging"):
+    """Search through original config to find default metrics for a given evaluation."""
+    if not original_config or not env or env not in original_config:
+        return None, None
+
+    # Look through all categories in the config
+    if "evaluations" in original_config[env]:
+        for category, evals_list in original_config[env]["evaluations"].items():
+            for eval_config in evals_list:
+                if eval_config.get("name") == eval_name:
+                    return (
+                        eval_config.get("default_scorer", "choice"),
+                        eval_config.get("default_metric", "accuracy"),
+                    )
+    return None, None
+
+
+def get_scores_from_file(path):
+    """Download and extract scores from the dashboard file."""
+    try:
+        s3 = boto3.client("s3")
+        bucket = os.environ.get("AWS_S3_BUCKET", "inspect-evals-dashboard")
+        response = s3.get_object(Bucket=bucket, Key=path)
+        data = json.loads(response["Body"].read().decode("utf-8"))
+
+        if "results" in data and "scores" in data["results"]:
+            return data["results"]["scores"]
+    except Exception:
```

Comment:
would some logging be useful here since you're not raising a specific exception?

Reply:
Done